### DISCUSSION TITLE
Meaning of fields in exported JSON

### DISCUSSION THREAD

**Discussion by smolck** [Chlumsky/msdf-atlas-gen#17] [Q&A] [answered]
In trying to use this library (just the binary for now), I've been successful in getting some things rendered from a generated font atlas, but I'm running into some issues that I think stem at least in part from not knowing what the different values in the exported JSON (from the file outputted when running the binary with `-json <output.json>`) mean.

For reference, here's part of the JSON for the font atlas I'm working with atm:
```json
{
  "atlas": {
    "type": "msdf",
    "distanceRange": 2,
    "size": 33.0625,
    "width": 104,
    "height": 104,
    "yOrigin": "bottom"
  },
  "metrics": {
    "emSize": 1,
    "lineHeight": 1.2090000000000001,
    "ascender": 0.96799999999999997,
    "descender": -0.24099999999999999,
    "underlineY": -0.20100000000000001,
    "underlineThickness": 0.095000000000000001
  },
  "glyphs": [
    {
      "unicode": 72,
      "advance": 0.746,
      "planeBounds": {
        "left": 0.025173913043478272,
        "bottom": -0.044817580340264687,
        "right": 0.72082608695652173,
        "top": 0.7718175803402646
      },
      "atlasBounds": {
        "left": 36.5,
        "bottom": 8.5,
        "right": 59.5,
        "top": 35.5
      }
    }
  ]
}
```
The values in the `atlas` and `metrics` objects I'm not so confused about (although an explanation or link to one of what exactly they mean/what units they're in, etc., would be helpful), but I don't really know what the `advance`, `planeBounds`, and `atlasBounds` fields in each of the objects in `glyphs` mean, what their units are, etc.? I've tried to figure it out through looking at the source code and also an exported shadron script, but I'm still unsure.

Thanks!

**Comment by Chlumsky** [comment, marked as answer]
See https://github.com/Chlumsky/msdf-atlas-gen/issues/2#issuecomment-636457340. The units are pixels for `size` and `atlasBounds` and EMs for everything else. Feel free to ask more specifically if something still isn't clear.

**Reply by smolck** [reply]
Ah, I didn't see that; thank you!
